### PR TITLE
Add AtenOp at:bitwise_or

### PR DESCRIPTION
**Pull request by ganik**
Support torch::bitwise_or training with ATenOp

### CODE REVIEW

File: onnxruntime/python/tools/symbolic_shape_infer.py [hunk 1]
```diff
@@ -189,6 +189,7 @@ def __init__(self, int_max, auto_merge, guess_output_rank, verbose, prefix=''):
         }
         self.aten_op_dispatcher_ = {
             'aten::embedding': self._infer_Gather,
+            'aten::bitwise_or': self._infer_aten_bitwise_or,
```
Comment (ganik) [inline onnxruntime/python/tools/symbolic_shape_infer.py:192]: @SherlockNoMad is this (shape inference for the bitwise_or op) necessary ?

Comment (SherlockNoMad) [reply onnxruntime/python/tools/symbolic_shape_infer.py:192]: yes

File: onnxruntime/python/tools/symbolic_shape_infer.py [hunk 2]
```diff
@@ -1075,6 +1076,17 @@ def _infer_Pool(self, node):
                 helper.make_tensor_value_info(o, vi.type.tensor_type.elem_type,
                                               get_shape_from_sympy_shape(sympy_shape)))
 
+    def _infer_aten_bitwise_or(self, node):
+        sympy_shape = self._get_sympy_shape(node, 0)
+        other = self._try_get_value(node, 1)
```
Comment (ashbhandare) [inline onnxruntime/python/tools/symbolic_shape_infer.py:1081]: why do we need to get the other value of input

File: onnxruntime/python/tools/symbolic_shape_infer.py [hunk 3]
```diff
@@ -1075,6 +1076,17 @@ def _infer_Pool(self, node):
...
+        assert other is not None
+
+        if node.output[0]:
+            vi = self.known_vi_[node.output[0]]
+            vi.CopyFrom(
```
Comment (ashbhandare) [inline onnxruntime/python/tools/symbolic_shape_infer.py:1086]: can you check if this op is supposed to support broadcasting? if yes, we need to infer output shape from both of the input shapes

File: orttraining/orttraining/python/training/ortmodule/_custom_op_symbolic_registry.py [hunk 4]
```diff
@@ -72,6 +72,10 @@ def embedding(g, weight, indices, padding_idx, scale_grad_by_freq, sparse):
         output.setType(output_type)
     return output
 
+@register_symbolic('bitwise_or')
+def bitwise_or(g, self, other):
+    return g.op("com.microsoft::ATenOp", self, other,
```
Comment (ashbhandare) [inline orttraining/orttraining/python/training/ortmodule/_custom_op_symbolic_registry.py:77]: can you check in export of both of the inputs are tensor? Alternatively, you can make this call into different overloads according to input types.

**Review by ashbhandare — CHANGES_REQUESTED** [review]

File: onnxruntime/python/tools/symbolic_shape_infer.py (hunk 2 above)
Comment (ganik) [reply onnxruntime/python/tools/symbolic_shape_infer.py:1081]: For the assert below. It has to present and be not None

File: orttraining/orttraining/python/training/ortmodule/_custom_op_symbolic_registry.py (hunk 4 above)
Comment (ganik) [reply orttraining/orttraining/python/training/ortmodule/_custom_op_symbolic_registry.py:77]: Yes both are Tensors

File: onnxruntime/python/tools/symbolic_shape_infer.py (hunk 3 above)
Comment (ganik) [reply onnxruntime/python/tools/symbolic_shape_infer.py]: > can you check if this op is supposed to support broadcasting? if yes, we need to infer output shape from both of the input shapes

Added broadcast in infer

**Review by baijumeswani — APPROVED** [review]

**Pull request merged**